src/engine/engineworkerscheduler.cpp

```diff
@@ -27,11 +24,17 @@ void EngineWorkerScheduler::workerReady(EngineWorker* pWorker) {
         // If the write fails, we really can't do much since we should not block
         // in this slot. Write the address of the variable pWorker, since it is
         // a 1-element array.
-        m_scheduleFIFO.write(&pWorker, 1);
+        m_workerStates[pWorker] = true;
```

Comment:
This needs at least a debug assertion that the worker has been registered before to avoid modifying QHash, But as already explained I'm not happy with the decision to use QHash here, at least not without additional safety lanes in form of many debug assertions that validate our assumptions.